flutter/documentation/result-spec.md

```diff
@@ -0,0 +1,129 @@
+and when fetching them from the online database.
+
+## General structure
+
+Results are serialized as JSON.
+
+Results must be a map with the following items in the root level:
+
+* `uuid`: string  
+UUID is generated by the app when all benchmarks are finished.
+* `results`: list of maps. See [List of benchmark-specific results](#benchmark-specific-results) section
+* `environment_info`: map. See [Environment info](#environment-info) section
+* `build_info`: map. See [Build info](#application-build-info) section
+
+## Benchmark-specific results
+
+Each run of a single benchmark generates one map with results.
+
+For example, if you select 3 benchmarks, you will get a list with 3 items, each describing specific benchmark run.
+
+If you additionally enable Submission mode, you will get a list of 6 items,
+because each benchmark will generate a performance and an accuracy mode result.
+
+* `benchmark_id`: string
+* `benchmark_name`: string  
+  Value from `task.model.name` for this benchmark from selected tasks.pbtxt file.
+* `loadgen_mode`: string enum  
+  See [`::mlperf::TestSettings`](https://github.com/mlcommons/inference/blob/08f5e36b74f4ec78ad738a287ae50462bb130330/loadgen/test_settings.h#L84).  
+  Allowed values:
+  * `accuracy_only`
+  * `performance_only`
+* `loadgen_scenario`: string enum  
+  See [`::mlperf::TestScenario`](https://github.com/mlcommons/inference/blob/a67f9f34bcc4439af4740095958c23380f9b284b/loadgen/test_settings.h#L38).  
+  Allowed values:
+  * `single_stream`
+  * `offline`
+* `dataset`: map  
+  Dataset info for this benchmark from selected tasks.pbtxt file.
+  * `name`: string
+  * `type`: string enum  
+    Allowed values:
+    * `IMAGENET`
+    * `COCO`
+    * `ADE20K`
+    * `SQUAD`
+  * `data_path`: string
+  * `groundtruth_path`: string
+* `backend_settings`: map  
+  Settings defined by selected backend for this benchmark.
+  * `accelerator_code`: string
+  * `accelerator_desc`: string
+  * `configuration`: string
+  * `model_path`: string
+* `backend_extra_settings`: list of maps  
+  Extra settings that can vary between different benchmarks and backends.  
+  Here must be stored values set by backend in `common_setting`.  
+  `batch_size` value for offline benchmarks should be located here.  
+  `shards_num` value for TFLite backend should be located here.  
+  Map structure:
+  * `id`: string. Value from `setting.id` that is passed to backend
+  * `name`: string. Value from `setting.name` that is passed to backend
+  * `value`: string. Value from `setting.value.value` that is passed to backend
+  * `value_name`: string. Value from `setting.value.name` that is passed to backend
+* `throughput`: floating point number. Present for all results
+* `accuracy`: floating point number. Present for all results, including `loadgen_mode=PerformanceOnly` runs  
+  When actual accuracy is not available value `-1` must be used
```

Comment:
Does JSON support [NaN](https://en.wikipedia.org/wiki/NaN)?

Reply:
NaN is not part of JSON specification.
It could still theoretically be supported by the tools/libs we use as an extension to JSON but it's not guaranteed.